### DISCUSSION TITLE
error in RPC call to 'Filecoin.ChainGetBlock': blockstore: block not found

### DISCUSSION THREAD

**Discussion by Sniker4** [filecoin-project/lotus#8200] [General Q&A] [answered]
Hi! I recently installed lotus and started using a lite node using the Protocol Labs api.chain.love Lotus full-node.

Calls from CLI like 'lotus chain getmessage' or 'lotus chain head' work smoothly, but I'm unable to get blocks from the chain.
Even if i do something like 'lotus chain head | xargs lotus chain getblock' i always receive :

WARN	rpc	go-jsonrpc@v0.1.5/handler.go:279	error in RPC call to 'Filecoin.ChainGetBlock': blockstore: block not found

What am I doing wrong/missing?

**Comment by LexLuthr** [comment, marked as answer]
@Sniker4 api.chain.love is a gateway managed by PL. The gateway actually run lotus-gateway and it exposes few APIs and not all of them. You can find the exposed API list at https://github.com/filecoin-project/lotus/blob/master/gateway/node.go
As you can see that lotus-gateway does not expose the API "Filecoin.ChainGetBlock" and thus you are seeing the error. The gateway is to get familiar with the lotus but you would need to run your own node to get full functionality.
